src/miral/minimal_window_manager.cpp

```diff
@@ -159,7 +167,14 @@ bool miral::MinimalWindowManager::handle_keyboard_event(MirKeyboardEvent const*
             return true;
 
         case KEY_TAB:
-            tools.focus_next_application();
+            if (self->application_selector.is_active())
+            {
+                self->application_selector.next();
+            }
+            else
+            {
+                self->application_selector.start(false);
+            }
```

Comment:
Asking the `application_selector` for its state to decide which `application_selector` method to call just looks weird.

Reply:
Agreed, this felt a bit wonky. In reality, you can call `next()` in any context and it would be valid.